tests/test_browser.go

```diff
@@ -259,6 +260,65 @@ func (b *testBrowser) await(fn func() error) error {
 	return b.vu.Loop.Start(fn)
 }
 
+// awaitWithTimeout is the same as await but takes a timeout and times out the function after the time runs out.
+func (b *testBrowser) awaitWithTimeout(timeout time.Duration, fn func() error) error {
```

Comment:
@mstoykov One thing I just noticed is that if a `testify/assert` or `require` fails somewhere in a Promise created in `fn`, then we would get the test failure, but would still have to wait for the timeout to expire, and would then get the timeout failure.

Is there a way of exposing the first error here, and aborting the timeout in that case?

Reply:
https://github.com/grafana/xk6-browser/pull/585